src/Angor/Client/Services/MessageService.cs

```diff
@@ -0,0 +1,326 @@
+        private string _contactHexPub;
+
+        public IReadOnlyList<DirectMessage> DirectMessages => _directMessages.AsReadOnly();
+        public bool IsLoadingMessages { get; private set; }
+        public bool IsSendingMessage { get; private set; }
+        public bool IsRefreshing { get; private set; }
+
+        public event Action OnChange;
+
+        public MessageService(
+            ILogger<MessageService> logger,
+            INostrCommunicationFactory nostrCommunicationFactory,
+            IEncryptionService encryptionService,
+            INetworkService networkService,
+            IRelayService relayService,
+            NostrConversionHelper nostrHelper)
+        {
+            _logger = logger;
+            _nostrCommunicationFactory = nostrCommunicationFactory;
+            _encryptionService = encryptionService;
+            _networkService = networkService;
+            _relayService = relayService;
+            _nostrHelper = nostrHelper;
+        }
+
+        public void SetKeys(string currentUserPrivateKeyHex, string currentUserNpub, string contactHexPub)
+        {
+            _currentUserPrivateKeyHex = currentUserPrivateKeyHex;
+            _currentUserNpub = currentUserNpub;
+            _contactHexPub = contactHexPub;
+
+            if (!string.IsNullOrEmpty(_currentUserNpub))
+            {
+                _currentUserHexPub = _nostrHelper.ConvertBech32ToHex(_currentUserNpub);
+            }
+            else
+            {
+                _currentUserHexPub = null;
+            }
+        }
+
+        public async Task InitializeAsync(string currentUserPrivateKeyHex, string currentUserNpub, string contactHexPub)
+        {
+            SetKeys(currentUserPrivateKeyHex, currentUserNpub, contactHexPub);
+
+            if (string.IsNullOrEmpty(_currentUserPrivateKeyHex) || string.IsNullOrEmpty(_contactHexPub) || string.IsNullOrEmpty(_currentUserHexPub))
+            {
+                _logger.LogWarning("InitializeAsync - Missing required keys.");
+                NotifyStateChanged();
+                return; 
+            }
+
+            IsLoadingMessages = true;
+            NotifyStateChanged();
+
+            try
+            {
+                await LoadMessagesAsync();
+                SubscribeToMessages();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Failed to initialize messaging: {ex.Message}");
+            }
+            finally
+            {
+                IsLoadingMessages = false;
+                NotifyStateChanged();
+            }
+        }
+
+        public async Task LoadMessagesAsync()
+        {
+            if (string.IsNullOrEmpty(_currentUserPrivateKeyHex) || string.IsNullOrEmpty(_contactHexPub) || string.IsNullOrEmpty(_currentUserHexPub))
+            {
+                _logger.LogWarning("LoadMessagesAsync - Missing required keys.");
+                return;
+            }
+
+            _directMessages.Clear();
```

Comment:
this is probably the issue, I suggest instead of clearing compare the events already there and only add the new ones.